File: benchmarks/oopsla25/subjects/263.py
import sys

def f263(s: str):
  c = s[len(s)-1]
  assert c == "b"
  if len(s) == 2:
    assert s[0] == "a"
  else:
    assert len(s) == 1


if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f263(my_string)

File: benchmarks/oopsla25/subjects/263.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f263(const char* s) {
    char c = s[strlen(s) - 1];
    assert(c == 'b');
    if (strlen(s) == 2) {
        assert(s[0] == 'a');
    } else {
        assert(strlen(s) == 1);
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f263(my_string);
  return 0;
}
